graphql-kotlin-client/src/main/kotlin/com/expediagroup/graphql/client/GraphQLClient.kt

```diff
@@ -90,13 +92,20 @@ class GraphQLClient<in T : HttpClientEngineConfig>(
     /**
      * Executes specified GraphQL query or mutation operation.
      */
-    suspend inline fun <reified T> execute(query: String, operationName: String? = null, variables: Any? = null): GraphQLResult<T> {
-        return executeOperation(query, operationName, variables, T::class.java)
+    suspend inline fun <reified T> execute(query: String, operationName: String? = null, variables: Any? = null): GraphQLResponse<T> {
+        return execute(query, operationName, variables, T::class.java)
+    }
+
+    /**
+     * Executes specified GraphQL query or mutation operation.
+     */
+    suspend inline fun <reified T> execute(request: GraphQLRequest): GraphQLResponse<T> {
```

Comment:
since those methods will be mostly executed through auto generated code -> do we need to have both (taking request object and individual params)?